src/Neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -384,5 +397,47 @@ protected internal void BurnGas(long gas)
                 throw new InvalidOperationException("GAS must be positive.");
             AddGas(gas);
         }
+
+        private static bool CheckItemType(StackItem item, ContractParameterType type)
+        {
+            StackItemType aType = item.Type;
+            if (aType == StackItemType.Pointer) return false;
+            switch (type)
+            {
+                case ContractParameterType.Any:
+                    return true;
+                case ContractParameterType.Boolean:
+                    return aType == StackItemType.Boolean;
+                case ContractParameterType.Integer:
+                    return aType == StackItemType.Integer;
+                case ContractParameterType.ByteArray:
+                case ContractParameterType.String:
```

Comment:
Don't we want to have some additional checks for the `String` type? The way it is now it's absolutely the same as `ByteArray`, maybe some differentiation can be useful like if `String` could only contain valid UTF-8 byte sequence.

Reply:
@shargon, how about this additional check?

Reply:
@AnnaShaleva  Done